### PR TITLE
Document OKRs

### PR DESCRIPTION
**Pull request by nicksnyder**

### CODE REVIEW

File: content/product/product_management/okrs.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+### Round-trip process
+
+The important property of OKR planning is that it is a complete round-trip with the team. If we don't do a round trip then it is possible for plans to not make business sense from a company level (if we only do bottom-up), or it is possible that the plan is infeasible (if we only do top-down). A round trip is also important because each person on the team has valuable context that we want to consider to reach the best plan.
+
+We choose to start our process top-down with a [direction document](#direction-document).
+
+Failure modes to avoid:
+
+- Orgs and teams accept direction and OKR drafts as given instead of a starting point for discussion. This failure mode is the equivalent of a top-down planning process with no round-trip.
```
Comment (jyavorska) [inline content/product/product_management/okrs.md:33]: Maybe another one is too many OKRs, which can result in lack of focus?

File: content/product/product_management/okrs.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Failure modes to avoid:
+
+- Engineering works on investments that aren't actually going to help us move faster toward short and long term goals.
+- Engineering doesn't proactively reserve time to work on necessary technical investments and our ability to deliver product improvements decreases over time.
+
+## Process
+
+### Direction document
+
+The OKR setting process starts with the VPs of Product and Engineering producing a document that describes the high level direction and strategy ([FY22-Q4 example](https://docs.google.com/document/d/1j_GuBOLXDVpzBzfCWJl-hz2rvqT--IPo3GYU07aG8HE/edit)).
```
Comment (jasonyavorsky) [inline content/product/product_management/okrs.md:50]: If what we do in that document as described in the heading https://docs.google.com/document/d/1j_GuBOLXDVpzBzfCWJl-hz2rvqT--IPo3GYU07aG8HE/edit#bookmark=id.tb35t1hjqf3z works, maybe we can bring some of those steps over here as official process.

File: content/product/product_management/okrs.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+OKRs are a contract between different management levels at our company.
+
+```
+CEO <-> Company OKRs <-> Executive team
+Product/Engineering VPs <-> Product/Engineering OKRs <-> Product/Engineering Directors
+Product/Engineering Directors <-> Product/Engineering org OKRs <-> Product/Engineering teams
+```
+
+### OKRs do not nest or cascade
```
Comment (jasonyavorsky) [inline content/product/product_management/okrs.md:19]: It doesn't exactly fit in this section, but doesn't fit in the others either.. maybe we should have something about not setting too many OKRs so we don't dilute focus?

File: content/product/product_management/okrs.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# Product and engineering OKRs
+
+## Philosophy
+
+### Product and engineering share OKRs
+
+Product and engineering have a shared goal to deliver a product that provides value to our customers, so we set shared OKRs.
+
+### OKRs are a contract
```
Comment (jasonyavorsky) [inline content/product/product_management/okrs.md:9]: As mentioned in standup, can we change this to something like "By setting objectives and key results, we are able to make our priorities visible across different departments and management levels at the company"? If OKRs become contracts they cease to be ambitious.

Comment (jasonyavorsky) [reply content/product/product_management/okrs.md:9]: Actually - double checking, do we set ambitious OKRs or ones that are treated like commitments that people are on the hook to hit and if they don't there are consequences? I've heard them described as both recently. Whichever it is, maybe good to write that down in this PR.

File: content/product/product_management/okrs.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+OKRs are a contract between different management levels at our company.
```
Comment (yinquanteo) [inline content/product/product_management/okrs.md:11]: This feels like it conflicts with [this](https://handbook.sourcegraph.com/company/goals/guidelines) 

The following basically says OKRs aren't meant to communicate commitments. Rather, they describe alignment around goals.
"Goals are for communication and alignment (everyone knowing what’s important, why, and how it’ll get done), not for estimation, scheduling, or promising/committing"

How do we reconcile this?

This ties into @jyavorska's question above as well.

File: content/product/product_management/okrs.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+CEO <-> Company OKRs <-> Executive team
+Product/Engineering VPs <-> Product/Engineering OKRs <-> Product/Engineering Directors
+Product/Engineering Directors <-> Product/Engineering org OKRs <-> Product/Engineering teams
+```
+
+### OKRs do not nest or cascade
+
+Each level of OKRs represent what is important at that level in the organization. This means that lower level OKRs do not nest under higher level OKRs, and objectives can be different at each level. Each layer of management is responsible for ensuring that the achievement of the OKRs one level below them will result in the achievement of the OKRs one level above them.
```
Comment (yinquanteo) [inline content/product/product_management/okrs.md:21]: I think it'll be helpful to include an example here just for illustrative purposes

File: content/product/product_management/okrs.md (hunk 1 above)
Comment (yinquanteo) [inline content/product/product_management/okrs.md:33]: 💯

File: content/product/product_management/okrs.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+Failure modes to avoid:
+
+- Orgs and teams accept direction and OKR drafts as given instead of a starting point for discussion. This failure mode is the equivalent of a top-down planning process with no round-trip.
+
+### Accounting for non-customer facing engineering work
+
+To sustain our ability to deliver over time, engineering is responsible for identifying, justifying, and prioritizing necessary technical investments (just like the product team is responsbile for identifying, justifying, and prioritizing product work). For big engineering initiatives, it may make sense to have a specific KR, but generally this kind of work can be thought of as an implementation detail of delivering on our product initiatives.
+
+This means that engineering doesn't need an explict KR to justify investment work; but they do need to be able to justify how this kind of work contributes toward our short and/or long term goals.
```
Comment (yinquanteo) [inline content/product/product_management/okrs.md:39]: Technically true, but in practice, specifying these sort of KRs has the effect of making specific engineering goals explicit. Which has a big effect in describing focus/importance which helps with morale/engagement/execution

File: content/product/product_management/okrs.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
+# Product and engineering OKRs
+
+## Philosophy
```
Comment (yinquanteo) [inline content/product/product_management/okrs.md:3]: One thing I'd love to add as a theme is – 
```
### Objectives should mostly not change quarter over quarter

Unless there are big shifts in company priorities or the underlying assumptions of the environment we're operating in, we'd generally expect Objectives to remain fairly constant even though the expected Key Results under them change.
```

File: content/product/product_management/okrs.md (hunk 3 above)
Comment (yinquanteo) [reply content/product/product_management/okrs.md:19]: agreed

**Comment by jasonyavorsky** [comment]
As @serinadawn mentioned in Slack, the content here is actually quite closely related to https://docs.google.com/document/d/1vYMH3s7B8mP2jRx5_9qYOiVJLs2ZX6T1i9NFHM1xgGk/edit#heading=h.i75peirn8gs6.

**Pull request closed without merging**